scripts/update-openapi-v2.sh

```
@@ -0,0 +1,52 @@
+  shift
+done
+
+echo "Fetching v2 OpenAPI spec from $API_URL..."
+
+# Create a temporary file for the raw response
+TEMP_FILE=$(mktemp)
+
+# Fetch the OpenAPI spec
+curl -s "$API_URL" > "$TEMP_FILE"
+
+if [ $? -ne 0 ]; then
```

Shellcheck (0.11.0)

[style] Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.

(SC2181)